### ISSUE TITLE
SDL Window disables composition under KWin by default

### ISSUE THREAD

**Issue by adamkruszewski** [let-def/texpresso#49]
Hello, first I would like to thank you for this great software! :)
Made editing LaTeX files in Emacs really pleasant experience.

The default behavior for SDL2 window is to disable composition if the Window Manager understands the hint. 
KDE's KWin does (probably is in the minority). The workaround is to either set the environment variable:
export SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR=0

or execute SDL_SetHint(SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR, "0");
before creating the window as someone mentioned here:  https://github.com/mosra/magnum/issues/184#issuecomment-425952900

Not sure what version of SDL you are targeting so I'm not expecting you'll add this line of code - just wanted to put it here so others may find it if they would like to workaround the issue with the env. variable.

Thanks,
Adam.

**Comment by let-def** [comment]
Hi,

Thanks for reporting this issue. I added the fix because I think it is more appropriate to keep composition enabled for an application like TeXpresso. I have only tested on recent Gnome/KDE with Wayland, not X11, so I never observed this problem.
The fix is now merged to the main branch, I believe it works (I just added the single line you mentioned, protected by a check for SDL2 compile time version).
